src/main/java/hudson/plugins/build_timeout/BuildStepWithTimeout.java

```diff
@@ -0,0 +1,153 @@
+package hudson.plugins.build_timeout;
+
+import hudson.Extension;
+import hudson.Launcher;
+import hudson.model.*;
+import hudson.plugins.build_timeout.impl.AbsoluteTimeOutStrategy;
+import hudson.plugins.build_timeout.impl.DeadlineTimeOutStrategy;
+import hudson.plugins.build_timeout.operations.AbortOperation;
+import hudson.tasks.BuildStep;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.BuildStepMonitor;
+import hudson.tasks.Builder;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
+import org.jenkinsci.plugins.tokenmacro.MacroEvaluationException;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.StaplerRequest;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+
+public class BuildStepWithTimeout extends Builder implements BuildStep {
+    private /* final */ BuildTimeOutStrategy strategy;
+    private BuildStep buildStep;
```

Comment:
To be honest, I think it's too dangerous :) 

From my perspective somebody should implement "multiple steps" build step once and all other plugins should just re-use it. I don't think that it's good idea to re-implement such logic, because even now I can't guarantee that all kind of build steps are supported (i.e. `conditional steps` doesn't work if it's used at child).

And in my case I don't need multiple steps - this "feature" was designed to split one global timeout on smallest chunks.